### PR TITLE
Add an issue template for Code Improvement

### PR DESCRIPTION
**Pull request by daniellepintz**
This will allow people to create an issue for a refactor or deprecation

## What does this PR do?

Currently if someone wants to make an issue for a refactoring or deprecation there is no option for this in the issue templates. This PR creates a template for a refactor/deprecation

Discussed in slack: https://app.slack.com/client/TR9DVT48M/C01296PBDQV/thread/C01296PBDQV-1629162951.025000

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by yifuwang** [comment]
This looks great! Thanks for proposing this.

Would it be better to call this "improvement" instead? IMHO refactoring is a means for improvement but itself is not an end.

**Comment by daniellepintz** [comment]
Good call @yifuwang how about "Code Improvement"? That encapsulates deprecation and refactoring

**Comment by yifuwang** [comment]
Thanks @daniellepintz! This sounds great

**Review by ananthsub — APPROVED** [review]
Looks reasonable to me! And glad that we're capturing these separately from bug reports and feature requests

**Review by s-rog — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — COMMENTED** [review]
feeling a bit confused, this looks almost the same as the feature request...

**Comment by daniellepintz** [comment]
@Borda you are right, it is almost the same. But this way when creating an issue there is an option for code improvement, as currently a lot of issues don't fall under the "feature" or "bug fix" category. One alternative is we could rename "Feature" to something to encapsulate feature, refactoring, and deprecation, but it might be a bit strange to just have a generic catch-all category. wdyt?

File: .github/ISSUE_TEMPLATE/code_improvement.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+name: Code improvement
+about: Suggest a code improvement, i.e. refactoring, deprecation, etc.
+title: ''
+labels: enhancement, help wanted
```
Comment (Borda) [inline .github/ISSUE_TEMPLATE/code_improvement.md:5]:
```suggestion
labels: enhancement, help wanted, refactors / code health
```

**Comment by daniellepintz** [comment]
@Borda @tchaton @edenlightning looks like I need a review from one of you to merge this. wdyt?

**Review by tchaton — APPROVED** [review]

**Pull request merged**